### PR TITLE
Flatten out subjects and remove duplicate subjects

### PR DESCRIPTION
**Pull request by kehh**

### CODE REVIEW

File: importScripts/refineryParameters/Subject.ca_objects.depicts.occurrenceSplitter.json [hunk 1]
```diff
@@ -1,10 +0,0 @@
-{
```
Comment (kehh) [inline importScripts/refineryParameters/Subject.ca_objects.depicts.occurrenceSplitter.json:1]: File no longer needed

File: importScripts/sampleRecords/subjectList.destination.json [hunk 2]
```diff
@@ -22,45 +22,64 @@
...
-      "idno":"PRINSEP, Henry Charles (Boncho",
+      "idno":"68",
+      "display_label":"Bussell, Josephine",
+      "ca_occurrences.type_id":"Subject",
+      "ca_occurrences.idno":"68",
+      "ca_occurrences.access":"accessible to public",
+      "ca_occurrences.preferred_labels":"Bussell, Josephine",
+      "ca_occurrences.LastEditBy":"IST",
+      "ca_occurrences.LastEditDate":"1 January 2000",
+      "ca_objects.idno":"A1986.138a;A1986.138b;A1986.138c;A1986.138d;A1986.138e;A1986.138f;A1986.138g;A1986.138I;A1986.138h;A1986.138j;A1986.138k;A1986.138l",
```
Comment (kehh) [inline importScripts/sampleRecords/subjectList.destination.json:33]: All these objects are now associated with Josephine Bussell

File: importScripts/sql/testImports/subjectList.sql [hunk 3]
```diff
@@ -16,13 +16,14 @@ UNION(SELECT s_SubjectID FROM SubjectsExport WHERE os_DateMax IS NOT NULL LIMIT
 UNION(SELECT s_SubjectID FROM SubjectsExport WHERE o_Accession_Full_ID IS NOT NULL LIMIT 2)
 UNION(SELECT s_SubjectID FROM SubjectsExport WHERE o_ItemType IS NOT NULL LIMIT 2)
 UNION(SELECT s_SubjectID FROM SubjectsExport WHERE ss_SubjectSubjectID IS NOT NULL LIMIT 2)
-UNION(SELECT s_SubjectID FROM SubjectsExport WHERE ss_SubjectID IS NOT NULL LIMIT 2)
-UNION(SELECT s_SubjectID FROM SubjectsExport WHERE ss_SubjectName IS NOT NULL LIMIT 2)
+UNION(SELECT s_SubjectID FROM SubjectsExport WHERE relatedSubject_SubjectID IS NOT NULL LIMIT 2)
```
Comment (kehh) [inline importScripts/sql/testImports/subjectList.sql:19]: Better name for the alias

File: importScripts/sql/testImports/subjectList.sql [hunk 4]
```diff
@@ -16,13 +16,14 @@ UNION(SELECT s_SubjectID FROM SubjectsExport WHERE os_DateMax IS NOT NULL LIMIT
...
+UNION(SELECT s_SubjectID FROM SubjectsExport WHERE relatedSubject_SubjectName IS NOT NULL LIMIT 2)
 UNION(SELECT s_SubjectID FROM SubjectsExport WHERE ss_Association IS NOT NULL LIMIT 2)
 UNION(SELECT s_SubjectID FROM SubjectsExport WHERE ss_Dates IS NOT NULL LIMIT 2)
 UNION(SELECT s_SubjectID FROM SubjectsExport WHERE ss_DateMin IS NOT NULL LIMIT 2)
 UNION(SELECT s_SubjectID FROM SubjectsExport WHERE ss_DateMax IS NOT NULL LIMIT 2)
-UNION(SELECT s_SubjectID FROM SubjectsExport WHERE rs_SubjectType IS NOT NULL LIMIT 2)
+UNION(SELECT s_SubjectID FROM SubjectsExport WHERE relatedSubject_SubjectType IS NOT NULL LIMIT 2)
+UNION(SELECT s_SubjectID FROM SubjectsExport WHERE s_SubjectName LIKE 'PRINSEP%' OR relatedSubject_SubjectName LIKE 'PRINSEP%' OR relatedSubject_SubjectName LIKE 'Bussel%' OR s_SubjectName LIKE 'Bussel%')
```
Comment (kehh) [inline importScripts/sql/testImports/subjectList.sql:26]: Extra sample records to match the problem names: https://youtrack.gaiaresources.com.au/youtrack/issue/RWAHS-147

**Comment by kehh** [comment]
Screenshots below show that there are only the two variants of PRINSEP for the sample records, not all the copies with IDNO `PRINSEP, Henry Charles (Boncho`
stagingmanyprinsep
noduplicatesubjects

File: importScripts/sql/testImports/subjectList.sql (hunk 4 above)
Comment (leftclickben) [inline importScripts/sql/testImports/subjectList.sql:26]: Is this all of the names that will cause the issue?  Maybe run a query to find `SubjectName` with length > the length of the `idno`?

**Comment by leftclickben** [comment]
Just the question about coverage.

File: importScripts/sql/testImports/subjectList.sql (hunk 4 above)
Comment (kehh) [inline importScripts/sql/testImports/subjectList.sql:26]: OK I can do. Looks like it is:

| SubjectName | len |
| --- | --- |
| Royal Western Australian Historical Society | 43 |
| PRINSEP, Henry Charles ("Bonchop") | 34 |
| PRINSEP, Henry Charles (Bonchop] | 32 |
| University of Western Australia | 31 |

**Comment by kehh** [comment]
OK I've added extra subjects, and all good:
onlyrealduplicates

**Pull request merged**